presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java

```diff
@@ -111,10 +115,24 @@ protected SchemaTableName getSchemaTableName(ResultSet resultSet)
     @Override
     protected String toSqlType(Type type)
     {
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            if (varcharType.getLength() <= MAX_IN_ROW_VARCHAR_LENGTH) {
```

Comment:
Poor idea imo. As depending on the columns set different type mapping would be used.
Let's keep it as it is as we discussed offline.